Add metadata.md to the HCMS documentation (#4093)
kyma-project/kyma: docs/headless-cms/03-03-markdown-documents.md

---
title: Markdown documents
type: Details
---

The DocsTopic or ClusterDocsTopic CR supports various documentation formats, including Markdown (`.md`) documents. Each `.md` file must consist of two parts: metadata and content. The following example illustrates the required structure: 

```
---
metadata: {value or text string}
another_metadata: {value or text string}
---

> {The content of your document in **Markdown**.}

Reviewer: ```suggestion
{The content of your document in Markdown.}
```

```

## Metadata

Each Markdown document displayed in the Kyma Console requires metadata in a specific format called [Front Matter](https://forestry.io/docs/editing/front-matter/). 

### Structure

When you create a Markdown document, define its `title` and `type`. Place the metadata at the top of your `.md` file, and separate it with three dashes.

```
---
title: {Document title}
type: {Document type}
---
```

- The `title` metadata defines the title of your document. 

> - The `type` metadata groups single documents together. Multiple documents that use the same `type` generate a grouping. For example, if you have multiple tutorials, you can group them below a navigation node called **Tutorials**.

Reviewer: ```suggestion
- The `type` metadata groups single documents together. Multiple documents that use the same `type` generate a grouping. For example, if you have multiple tutorials, you can group them under a navigation node called **Tutorials**.
```

> >**NOTE:** If there is only one document of a certain type, remove the `type` metadata, so that the document displays well in the UIs.

Reviewer: ```suggestion
>**NOTE:** If there is only one document of a certain type, remove the `type` metadata, so that the document displays well in the UI.
```

### Display

> In the Docs UI, which is the view displayed once you click on the question mark icon ![](./assets/docs-ui-question-mark.png) in the top-right corner of the Console, the metadata help create the left-side navigation structure. The Docs UI displays documents grouped under a common `type` in alphanumeric order as per files names. The following example shows four documents, their metadata, and corresponding places in the left-side navigation.

Reviewer: ```suggestion
In the Docs UI, which is the view displayed once you click on the question mark icon ![](./assets/docs-ui-question-mark.png) in the top-right corner of the Console, the metadata fields help to create the left-side navigation structure. The Docs UI displays documents grouped under a common `type` in alphanumeric order as per files names. The following example shows four documents, their metadata, and corresponding places in the left-side navigation.
```

Author: ```suggestion
In the Docs UI, which is the view displayed once you click on the question mark icon ![](./assets/docs-ui-question-mark.png) in the top-right corner of the Console, the metadata allow you to create the left-side navigation structure. The Docs UI displays documents grouped under a common `type` in alphanumeric order as per files names. The following example shows four documents, their metadata, and corresponding places in the left-side navigation.
```

<div tabs>
  <details>
  <summary>
  Metadata source
  </summary>

```
//03-01-sidecar-proxy-injection.md
---
title: Sidecar Proxy Injection
type: Details
---
``` 
```
//03-02-istio-patch.md
---
title: Istio Patch
type: Details
---
```
```
//03-03-istio-rbac.md
---
title: Istio RBAC configuration
type: Details
---
```
```
//01-01-service-mesh.md
---
title: Overview
---
```
  </details>
  <details>
  <summary>
  Docs UI preview
  </summary>

![](./assets/left-side-nav-docs-ui.png)
  </details>
</div>

> In the Service Catalog and Instances views, which contain Service Classes documentation, the `title` metadata displays as the name of a particular tab. If you don't provide `title`, the UIs display the file name as a fallback. See the following example:

Reviewer: ```suggestion
In the Service Catalog and Instances views, which contain Service Classes documentation, the `title` metadata displays as the name of a particular tab. If you don't provide `title`, the UI displays the file name as a fallback. See the following example:
```

<div tabs>
  <details>
  <summary>
  Metadata source
  </summary>

```
//overview.md
---
title: Overview
---
``` 
```
//plan-details.md
---
title: Services and Plans
type: Details
---
```
  </details>
  <details>
  <summary>
  Service Catalog preview
  </summary>

![](./assets/tabs-catalog-ui.png)
  </details>
</div>

> >**NOTE:** A document with `title` **Overview** always displays as the first tab. Markdown files with `title` other than **Overview** appear in alphanumeric order.

Reviewer: ```suggestion
>**NOTE:** A document with the **Overview** `title` always displays as the first tab. Markdown files with `title` other than **Overview** appear in alphanumeric order.
```

## Content

Content is the body of your document. Write content in [Markdown](https://daringfireball.net/projects/markdown/syntax) which is a simplified markup language.

In Kyma, to make documentation more reader-friendly, some Markdown features are customized. See the following examples:

> 1. Linking - link between documents in the same topic or in different topics using metadata. 

Reviewer: Maybe instead of providing examples in this document, it would be better to link to our guidelines? [Here](https://github.com/kyma-project/community/blob/master/guidelines/content-guidelines/links-in-docs.md) is the document describing the links in docs. Same comment for the toggle and the panels. Let's discuss.
